tiatoolbox/tools/registration/wsi_registration.py

```diff
@@ -103,6 +119,24 @@ def prealignment(
     fixed_img = exposure.rescale_intensity(img_as_float(fixed_img), in_range=(0, 1))
     moving_img = exposure.rescale_intensity(img_as_float(moving_img), in_range=(0, 1))
 
+    # Resizing of fixed and moving masks so that dice can be computed
+    height = np.max((fixed_mask.shape[0], moving_mask.shape[0]))
+    width = np.max((fixed_mask.shape[1], moving_mask.shape[1]))
+    padded_fixed_mask = np.pad(
+        fixed_mask,
+        pad_width=[(0, height - fixed_mask.shape[0]), (0, width - fixed_mask.shape[1])],
+        mode="constant",
+    )
+    padded_moving_mask = np.pad(
+        moving_mask,
+        pad_width=[
+            (0, height - moving_mask.shape[0]),
+            (0, width - moving_mask.shape[1]),
+        ],
+        mode="constant",
+    )
+    dice_before = dice(padded_fixed_mask, padded_moving_mask)
+
     cy, cx = ndi.center_of_mass((1 - fixed_img) * fixed_mask)
```

Comment:
A minor point, but this may be fast with cv2.
See https://stackoverflow.com/questions/49582008/center-of-mass-in-contour-python-opencv

Reply:
I have replaced scipy with OpenCV.  I found scipy straightforward for the computation of center of mass as compared to opencv. 
https://github.com/TissueImageAnalytics/tiatoolbox/blob/763922b6e2c5d58ffde77d4eada2d15607751fd2/tiatoolbox/tools/registration/wsi_registration.py#L62-L80